MyFirstSlideshow/app/src/main/res/layout/activity_main.xml

```diff
@@ -4,11 +4,87 @@
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     tools:context="com.charles.myfirstslideshow.MainActivity">
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:orientation="vertical">
 
-    <TextView
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:text="Hello World!"
```

Comment:
activity_main: line 11 - keep the text in the strings.xml and gave its path here.

Reply:
activity_main: line 11 - we should keep all the strings in res/strings.xml file instead of using it directly. it will be safe to make typo error. and memory use also